webhooky/src/handlers_checkr.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use cio_api::{companies::Company, db::Database};
+use dropshot::{Extractor, RequestContext, ServerContext};
+use dropshot_auth::sig::HmacSignatureVerifier;
+use hmac::Hmac;
+use log::info;
+use sha2::Sha256;
+use std::{borrow::Cow, sync::Arc};
+
+use crate::http::Headers;
+
+#[derive(Debug)]
+pub struct CheckrWebhookVerification;
+
+#[async_trait]
+impl HmacSignatureVerifier for CheckrWebhookVerification {
+    type Algo = Hmac<Sha256>;
+
+    async fn key<'a, Context: ServerContext>(_: &'a Arc<RequestContext<Context>>) -> Result<Cow<'a, [u8]>> {
+        // We only have a generic context here so we can not take values out. Instead construct a
+        // new db connection in the meantime
+        let db = Database::new().await;
+
+        Ok(Company::get_from_db(&db, "Oxide".to_string())
```

Comment:
It seems like the other verifiers all pull their keys out of the environment, but this one uses the database. Is that maybe why the checkr tests are disabled below? Could we check the environment first, and fall back to the db? I don't really know anything about the context in which this code runs, so apologies if that doesn't really make sense.

Reply:
Updated this to check the environment first so that tests can be implemented. There are some historical reasons for reading it from the db I believe.